api/glossary_usefulconcepts.md

```diff
@@ -0,0 +1,156 @@
+
+These can all be easily Googled for further clarity and descriptions as to how they modify the respective functions.
+
+## Endpoints
+The [API endpoint](https://smartbear.com/learn/performance-monitoring/api-endpoints/) is the end of a communication channel. When there are two interacting APIs, these interact with each other at the endpoint. This is the location from which APIs can access the network resources they need to carry out their functions.
+APIs work using `requests` and `responses` in that they request information from a web application or web server and then receive a response. Endpoints send the requests and indicate where the resource lives. These are stored in the [routes.py](https://github.com/deep-learning-indaba/Baobab/blob/develop/api/app/routes.py) file in the [Baobab repository](https://github.com/deep-learning-indaba/Baobab).
+
+## Models
+These are an example of a Python object, specifically a database.
+
+## Relational databases
+The type of database we are interested in is called a relational database (and this is why ORM [Object Relational Mapping] is so relevant).
+Relational databases set up relationships between items efficiently, and each data are stored in a table. The rows contain the items and the columns contain the properties of the items. The ORM (in our case, SQLAlchemy) is written in a language that is part of the web framework language (in our case, Python), allowing us to communicate with the database that is more familiar than SQL.
+
+####Organisation of the data in a database versus a class:
+**Class:**
+
+|ClassName|
+|---------|
+|**id:** int|
+|**name:** str|
+|**descriptions:** str|
+
+**Relational database:**
+
+|ID:|Name:|Description:|
+|---|---|---|
+|INT|Char|VarChar|
+
+### Database keys
+Keys are fundamental in the creation of relationships between data (intra- and inter-tables). Further, they assist in maintaining uniqueness in a table; ensure consistent and valid data as well as allow for efficient data retrieval.
+
+|Candidate keys (multiple, one can be a primary key)||
+|--------------------------------------|---|
+|**Primary keys**|**Alternate key:**|
+|Uniquely identifies each record. Cannot be NULL.|Currently not selected as a primary key|
+|Ensures entity integrity (ensures there are no duplicate records within the table)|**Unique key:** |
+| |Similar to primary keys, but all for one null value|
+| |No duplicates, useful for data validation|
+| |Useful to place additional unique conditions on columns|
+|**Composite keys:**| |
+|a.k.a. Compound, composite keys| |
+| More than one columns used to identify a unique value|
+
+### Data types
+These define the type of value that can be stored in a column. 
+This summary of [PostgreSQL data types](https://www.postgresql.org/docs/9.5/datatype.html) will provide more information.
+- Numeric: 
+    - float, real, integers (no decimal, varying precision), decimal (fixed precision (p,s))
+- Characters:
```

Comment:
Rather say "strings" which I think will be more familiar to Python programmers..